### PR TITLE
feat(protocol-designer): refactor and restyle LabwareSelectionModal

### LINKED ISSUE #1781: Changes to labware selection modal
## Background
- Currently when you click "add labware" the labware selection modal can appear pretty far away from the slot you've selected. Screenshot:
image
- When you click outside of the modal the modal doesn't close. 
- These two combined created confusion in a usertest about what was going on

## Implementation notes
- Labware select modal is a NEW component. 
- Component is full height but not full width
- Modal scrolls if content exceeds height
- Does not come with an overlay

## Acceptance criteria
- Create new labware select modal component. Aside from spacing it's styled the same as our other modal. 
- Modal will go in the center bc that's easier than putting it near the selected slot
- Selected slot number is reinforced in title of modal
- When clicking outside of the modal, modal should close

## Design
- Didn't draw every list item for all our labware, but if you want I can go back and do that
screen shot 2018-07-05 at 12 20 53 pm

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #1781 

## changelog

- rename, refactor and restyle LabwareSelectionModal

## review requests

- Labware modal adds labware, looks nice (except for the hover images, they're kinda weird)
- Labware modal closes when you click the CLOSE button, or when you click off of it
- Code is cleaner

### CODE REVIEW

File: protocol-designer/src/components/IngredientsList.css [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @import '@opentrons/components';
 
 .close_icon {
-  & svg {
+  & > svg {
```
Comment (IanLondon) [inline protocol-designer/src/components/IngredientsList.css]: re: @mcous comment in prev PR

File: protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  onClose: (e?: SyntheticEvent<*>) => mixed,
+  selectLabware: selectLabware,
+  slot: ?DeckSlot,
+  permittedTipracks: Array<string>,
+  select: (?string) => mixed,
+  selectedSection: ?string
+}
+
+// labware type, display name, and image url
+const hardcodedLabware = {
```
Comment (IanLondon) [inline protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js]: soon we'll use shared-data and do something else with the tooltip images, but for now, I'm only doing display stuff in this PR

File: protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      labwareInSection = labwareInSection.filter(labwareModelNameImage =>
+        permittedTipracks.includes(labwareModelNameImage[0])
+      )
+    }
+    return (
+      <PDTitledList
+        key={section}
+        title={section}
+        collapsed={selectedSection !== section}
+        // TODO IMMEDIATELY: use a maker for these fns
```
Comment (IanLondon) [inline protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js]: TODO

File: components/src/styles/borders.css [hunk 4]
```diff
@@ -1,8 +1,26 @@
...
+  --outline-highlight: {
+    border-color: transparent;
+    outline: 2px solid var(--c-highlight);
+    outline-width: 2px 0;
+  }
+
+  /* dropshadows */
+
+  /* TODO: Ian 2018-07-26 consolidate all shadows here (eg from buttons) */
+  --shadow-lvl-1:
```
Comment (IanLondon) [inline components/src/styles/borders.css]: lvl 1 and lvl 2 names of dropshadows are from zeplin, btw

File: protocol-designer/src/components/LabwareSelectionModal/styles.css [hunk 5]
```diff
@@ -0,0 +1,47 @@
+@import '@opentrons/components';
```
Comment (IanLondon) [inline protocol-designer/src/components/LabwareSelectionModal/styles.css]: from deleted `LabwareDropdown.css` with some changes

File: protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+          permittedTipracks.includes(labwareModelNameImage[0])
+        )
+      }
+
+      return (
+        <PDTitledList
+          key={section}
+          title={section}
+          collapsed={selectedSection !== section}
+          onCollapseToggle={selectSection}
```
Comment (b-cooper) [inline protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js]: This isn't actually toggling at the moment. If you click an open section it doesn't unselect.

File: protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    return sections
+  }
+
+  render () {
+    const {onClose, slot} = this.props
+    // do not render without a slot
+    if (!slot) return null
+
+    return (
+      <ClickOutside onClickOutside={onClose}>
```
Comment (b-cooper) [inline protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js]: 👶 as a function!

File: protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js (hunk 6 above)
Comment (IanLondon) [reply protocol-designer/src/components/LabwareSelectionModal/LabwareSelectionModal.js]: @howisthisnamenottakenyet should open sections be collapsible in this modal?

**Pull request merged**